client/src/components/App.jsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import $ from 'jquery';
+import Calendar from 'react-calendar';
+import ClickOutHandler from 'react-onclickout';
+import StarRating from './StarRating.jsx';
+import MyCalendar from './MyCalendar.jsx';
+import Guests from './Guests.jsx';
+import PricingTotal from './PricingTotal.jsx';
+
+class App extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      showCalendar: false,
+      showGuests: false,
+      listingId: props.listingId,
+      listingInfo: {},
+      adults: 1,
+      children: 0,
+      infants: 0,
+      startDate: null,
+      endDate: null
+    };
+    this.handleGuestsClick = this.handleGuestsClick.bind(this);
+    this.updateGuestsTotal = this.updateGuestsTotal.bind(this);
+    this.triggerPricingTotal = this.triggerPricingTotal.bind(this);
+  }
+
+  componentDidMount() {
+    this.getBookingInfo(this.state.listingId);
+  }
+
+  handleGuestsClick(boolean) {
+    this.setState({
+      showGuests: boolean
+    });
+  }
+
+  getBookingInfo() {
+    $.ajax({
+      method: 'GET',
+      url: `http://127.0.0.1:8080/rooms/${this.props.listingId}/bookings`,
+      success: (data) => {
+        console.log('Ajax success!');
+        this.setState({
+          listingInfo: data
+        });
+      },
+      error: (err) => {
+        console.log('Ajax error!', err);
```

Comment:
Would it be better to throw an error here instead of logging it? I am not sure when to do which